### PR TITLE
Add BlastApi on Optimism documentation

### PR DESCRIPTION
**Pull request by fmanea**
**Description**
We have added Optimism Mainnet dedicated API support and Public API support on :
https://blastapi.io/
https://blastapi.io/public-api

We also wanted these features to be reflected in the documentation, so developers can use them right away:
https://twitter.com/optimismPBC/status/1529175842185682951

### CODE REVIEW

File: src/docs/useful-tools/networks.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ We recently removed our whitelist! This means that anyone can deploy onto Optimi
 | Description | **`Mainnet`** |
 | Chain ID | **`10`** |
 | Explorer | **`https://optimistic.etherscan.io`** |
-| HTTP Endpoint | **`https://mainnet.optimism.io`** _Not for production systems._ See the list of available RPC endpoints below |
-| WebSocket Endpoint | **`wss://ws-mainnet.optimism.io`** |
+| HTTP Endpoint | **`https://mainnet.optimism.io`** **`https://optimism-mainnet.public.blastapi.io`** _Not for production systems._   See the list of available RPC endpoints below |
+| WebSocket Endpoint | **`wss://ws-mainnet.optimism.io`** **`wss://optimism-mainnet.public.blastapi.io`** |
```
Comment (qbzzt) [inline src/docs/useful-tools/networks.md:19]: Are these endpoints that anybody can use without any authentication? In general we use these rows for our endpoints.

**Review by qbzzt — CHANGES_REQUESTED** [review]

File: src/docs/useful-tools/networks.md (hunk 1 above)
Comment (fmanea) [reply src/docs/useful-tools/networks.md:19]: These endpoints can be used without any authentication. They are public, each endpoint is rate limited at a generous 25 req/s limit per IP.

Comment (fmanea) [reply src/docs/useful-tools/networks.md:19]: Do you recommend adding this information to a new section? something like external Public APIs providers?

Comment (fmanea) [reply src/docs/useful-tools/networks.md]: @qbzzt - removed those changes as requested in the next commit.

**Comment by qbzzt** [comment]
The PR looks good, but I'm afraid when I tried your endpoint it didn't work. I got this error:

```
  reason: 'bad response',
  code: 'SERVER_ERROR',
  status: 400,
  headers: {
    date: 'Fri, 27 May 2022 17:11:33 GMT',
    'content-type': 'application/json',
    'content-length': '145',
    connection: 'close',
    'strict-transport-security': 'max-age=15724800; includeSubDomains',
    'access-control-allow-origin': '*',
    'access-control-allow-credentials': 'true',
    'access-control-allow-methods': 'GET, PUT, POST, DELETE, PATCH, OPTIONS',
    'access-control-allow-headers': 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization',
    'access-control-max-age': '1728000'
  },
  body: `{"jsonrpc":"2.0","id":44,"error":{"code":-32601,"message":"Method not found","data":{"method": "'eth_getLogs' does not exist/is not available"}}}`,
  requestBody: '{"method":"eth_getLogs","params":[{"fromBlock":"0x0","toBlock":"latest","address":"0x4200000000000000000000000000000000000007","topics":["0x4641df4a962071e12719d8c8c8e5ac7fc4d97b927346a3d7a335b1f7517e133c","0x717558f6cca9e0f1ffc0f053d83171ec74a5a8f2a20594234b5c6c8f63b14964"]}],"id":44,"jsonrpc":"2.0"}',
  requestMethod: 'POST',
  url: 'https://optimism-mainnet.public.blastapi.io'
}
```

Can you fix it and then we'll merge?

**Comment by fmanea** [comment]
The endpoint is working but due to the fact that the eth_getLogs call is an intensive process, this call is not enabled in our Public APIs, but all the other method calls are available. 

The eth_getLogs is available on our Dedicated endpoint offering. You can obtain one by connecting with Metamask.

I can share the documentation about the PublicAPI next week, if that helps.

I hope this helps,
Flavian

**Comment by qbzzt** [comment]
> I can share the documentation about the PublicAPI next week, if that helps.

Yes, please do. We look bad if we recommend something and it doesn't work, but if we say "it works except for methods ..." then we're all good.

**Review by qbzzt — APPROVED** [review]
Sorry for taking so long - the last few days have been ... busy.

**Comment by fmanea** [comment]
No other methods. But I think this was merged and closed already.

**Pull request merged**